LkeServices/Blockchain/SiriusWalletsService.cs

```diff
@@ -36,189 +39,377 @@ public class SiriusWalletsService : ISiriusWalletsService
+                context: new { accountId, clientId, assetId });
+
+            return null;
+        }
+
+        public async Task<bool> IsAddressValidAsync(string blockchainId, string address)
+        {
+            var response = await _siriusApiClient.Addresses.IsValidAsync(new AddressIsValidRequest
+            {
+                Address = address, BlockchainId = blockchainId
+            });
+
+            return response.ResultCase == AddressIsValidResponse.ResultOneofCase.Body && response.Body.IsFormatValid;
+        }
 
-                    if (user.BodyCase == CreateUserResponse.BodyOneofCase.Error)
+        public async Task<List<BlockchainResponse>> GetBlockchainsAsync()
+        {
+            var response = await _siriusApiClient.Blockchains.SearchAsync(new BlockchainSearchRequest());
+
+            return response.ResultCase == BlockchainSearchResponse.ResultOneofCase.Body
+                ? response.Body.Items.ToList()
+                : new List<BlockchainResponse>();
+        }
+
+        private async Task<AccountSearchResponse> SearchAccountAsync(string clientId)
+        {
+            var retryPolicy = Policy
+                .Handle<Exception>(ex =>
+                {
+                    _log.WriteWarning(nameof(SearchAccountAsync), new { clientId }, $"Retry on Exception: {ex.Message}.", ex);
+                    return true;
+                })
+                .OrResult<AccountSearchResponse>(response =>
                     {
-                        _log.WriteWarning(nameof(CreateWalletsAsync), info: "Error creating user in sirius", context: new { error = user.Error, clientId, requestId = userRequestId });
-                        return;
+                        if (response.ResultCase == AccountSearchResponse.ResultOneofCase.Error)
+                        {
+                            _log.WriteWarning(nameof(SearchAccountAsync), response.ToJson(), "Response from sirius.");
+                        }
+
+                        return response.ResultCase == AccountSearchResponse.ResultOneofCase.Error;
                     }
+                )
+                .WaitAndRetryAsync(_delay);
+
+            try
+            {
+                var result = await retryPolicy.ExecuteAsync(async () => await _siriusApiClient.Accounts.SearchAsync(new AccountSearchRequest
+                {
+                    BrokerAccountId = _brokerAccountId, UserNativeId = clientId, ReferenceId = clientId
+                }));
 
-                    _log.WriteInfo(nameof(CreateWalletsAsync), info: "Creating wallets in sirius", context: new { clientId, requestId = accountRequestId });
+                if (result.ResultCase != AccountSearchResponse.ResultOneofCase.Error)
+                    return result;
 
-                    var createResponse = await _siriusApiClient.Accounts.CreateAsync(new AccountCreateRequest
+                _log.WriteError(nameof(SearchAccountAsync), new { clientId });
```

Comment:
Add a message here to simplify the identification of the code that writes to the logs.